azurerm/internal/services/recoveryservices/resource_arm_backup_policy_vm.go

```diff
@@ -263,6 +264,11 @@ func resourceArmBackupProtectionPolicyVM() *schema.Resource {
 					return fmt.Errorf("`retention_daily` must be set when backup.0.frequency is daily")
 				}
 
+				// Less than 7 daily backups is no longer supported for create/update
+				if diff.Get("retention_daily.0.count").(int) < 7 {
```

Comment:
I wish we could set this up for users who are still using setups that are less than 7 but we have to make a best effort here. I don't think this section will work because this code will trigger regardless of an update or not and this will break someone who is using the old system. 

I think our best effort is to just prevent users from creating this resource with less than a count of 7. It'll be a little more work and won't prevent someone from trying to update `count` to less than 7 but it'll make sure we don't break users that are currently using less than 7. 

That change is going to require a bit more work though. We'll want to update this [method](https://github.com/terraform-providers/terraform-provider-azurerm/blob/f0c4b1f503dba0b0ababd33d7626f218327a38d7/azurerm/internal/services/recoveryservices/resource_arm_backup_policy_vm.go#L487) by using `d.IsNewResource()` and we error out if it is a new resource with a count less than 7.

Let me know if that makes sense and if not, I can go into more detail.